core/src/main/shim/2.4/SparkShim.scala

```diff
@@ -158,4 +160,10 @@ object SparkShim extends SparkShimBase {
       }
     }
   }
+
+  def newUnresolvedException[TreeType <: TreeNode[_]](
+      tree: TreeType,
+      function: String): Exception = {
```

Comment:
leave out the function here?

Reply:
I think we need this to compile against Spark 2.4, right?